app-editors/atom/atom-1.27.2.ebuild

```diff
@@ -0,0 +1,563 @@
+}
+
+# Return the installation target directory.
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+# Return the Electron installation directory.
+get_electron_dir() {
+	echo -n "${EROOT%/}/usr/$(get_libdir)/electron-${ELECTRON_SLOT}"
+}
+
+# Return the directory containing appropriate Node headers
+# for the required version of Electron.
+get_electron_nodedir() {
+	echo -n "${EROOT%/}/usr/include/electron-${ELECTRON_SLOT}/node/"
+}
+
+# Run JavaScript using Electron's version of Node.
+enode_electron() {
+	"$(get_electron_dir)"/node $@
+}
+
+# Run node-gyp using Electron's version of Node.
+enodegyp_atom() {
+	local apmpath="$(get_atom_rpmdir)/resources/app/apm"
+	local nodegyp="${BIN_S}/${apmpath}/node_modules/node-gyp/bin/node-gyp.js"
+
+	PATH="$(get_electron_dir):${PATH}" \
+		enode_electron "${nodegyp}" \
+			--nodedir="$(get_electron_nodedir)" $@ || die
+}
+
+# Coffee Script wrapper.
+ecoffeescript() {
+	local cscript="${FILESDIR}/transpile-coffee-script.js"
+
+	# Disable shell glob expansion, as we want the coffee script
+	# transpiler to do that instead.
+	set -f
+	echo "ecoffeescript" $@
+	ATOM_HOME="${T}/.atom" ATOM_SRC_ROOT="${BUILD_DIR}/app" \
+	NODE_PATH="${BUILD_DIR}/app/node_modules" \
+		enode_electron "${cscript}" $@ || die
+	set +f
+}
+
+# asar wrapper.
+easar() {
+	local asar="${WORKDIR}/$(package_dir asar)/node_modules/asar/bin/asar"
+	echo "asar" $@
+	enode_electron "${asar}" $@ || die
+}
+
+# Return a $WORKDIR directory for a given package name.
+package_dir() {
+	local binmod="${1//-/_}"
+	local binmod_v="${binmod^^}_V"
+	echo -n ${1}-${!binmod_v}
+}
+
+# Check if there are patches for a given package.
+_have_patches_for() {
+	local _patches="${1}-*.patch" _find
+	_find=$(find "${FILESDIR}" -maxdepth 1 -name "${_patches}" -print -quit)
+	test -n "$_find"
+}
+
+# Tarballs on registry.npmjs.org are wildly inconsistent,
+# and violate the convention of having ${P} as the top directory name.
+# This helper detects and fixes that.
+_unpack() {
+	local a="${1}"
+	local b="${a%.tar.gz}"
+	local p="${b#atomdep-}"
+	local dir="$(tar -tzf "${DISTDIR}/${a}" | head -1 | cut -f1 -d'/')"
+
+	unpack "${a}"
+
+	if [[ "${dir}" != "${p}" ]]; then
+		# Set the correct name for the unpacked directory.
+		mv "${WORKDIR}/${dir}" "${WORKDIR}/${p}" || die
+	fi
+}
+
+# Check if the binary node module is actually a valid dependency.
+# Sometimes the upstream removes a dependency from package.json but
+# forgets to remove the module from node_modules.
+_is_valid_binmod() {
+	local mod
+
+	for mod in "${BINMODS[@]}"; do
+		if [[ "${mod}" == "${1}" ]]; then
+			return 0
+		fi
+	done
+
+	return 1
+}
+
+# Replace binary node modules with the newly compiled versions thereof.
+_fix_binmods() {
+	local _dir="${2}"
+	local _prefix="${1}"
+	local path
+	local relpath
+	local modpath
+	local mod
+	local f
+	local d
+	local cruft
+
+	(find "${_prefix}/${_dir}" -name '*.node' -print || die) \
```

Comment:
Use the friendly while-read-find loop you'll find in some eclasses.

Reply:
Can you elaborate please?  Most examples I could find put `find` at the end of the loop, i.e instead of

`find | while read do ...`

you get

`while read do < find`

I actually find the latter harder to read.  Is there a technical reason why the latter is preferred?